status/unstructured_adapter_test.go

```diff
@@ -35,7 +35,7 @@ var _ = Describe("Unstructured Adapter", func() {
 			Kind:    "testKind",
 		})
 
-		conditionObj := status.NewUnstructuredAdapter(testObject)
+		conditionObj := status.NewUnstructuredAdapter[*TestObject](testObject)
```

Comment:
musing: Don't do it in this PR but we may be able to get rid of passing the object into the adapter entirely now that the adapter itself carries the generic type